source/include/FreeRTOS_Routing.h

```diff
@@ -62,6 +67,25 @@
         NetworkInterfaceInitialiseFunction_t pfInitialise; /**< This function will be called upon initialisation and repeated until it returns pdPASS. */
         NetworkInterfaceOutputFunction_t pfOutput;         /**< This function is supposed to send out a packet. */
         GetPhyLinkStatusFunction_t pfGetPhyLinkStatus;     /**< This function will return pdTRUE as long as the PHY Link Status is high. */
+        #if ( ipconfigIS_ENABLED( ipconfigMAC_FILTERING ) )
+
+            /* The network driver is responsible for keeping track of which MAC addresses ( unicast or multicast )
```

Comment:
Your comment is still talking a lot about multicast, whereas the same logic applies to unicast.
I have a proposal:
~~~
/*
 * The EMAC probably has a 64-bit hash table. For every MAC-address
 * the hash value (mod 64) is calculated.  pfAddAllowedMAC() will set the bit,
 * and pfRemoveAllowedMAC() will clear the bit.  However, it may
 * happen that multiple MAC-addresses result in accessing the same bit.
 * A quick and dirty implementation option is to receive all MAC
 * addresses and set both pfAddAllowedMAC and pfRemoveAllowedMAC
 * to NULL, resulting in a promiscuous interface.
 * The optimal way to implement this functionality depends on the
 * specific hardware EMAC.
 * For one implementation option, check out:
 * "portable/NetworkInterface/DriverSAM/NetworkInterface.c".
 * It counts the number of times that a particular bit was set/cleared.
 * When the counter reaches zero, the has value will be disabled.
 */
~~~
Please correct my proposal, as your proficiency in English is surely better than mine.